lib/core/elastic_connector_actions.rb

```diff
@@ -37,6 +37,11 @@ def connectors_meta
         alias_mappings = client.indices.get_mapping(:index => Utility::Constants::CONNECTORS_INDEX).with_indifferent_access
         index = get_latest_index_in_alias(Utility::Constants::CONNECTORS_INDEX, alias_mappings.keys)
         alias_mappings.dig(index, 'mappings', '_meta') || {}
+      rescue Elastic::Transport::Transport::Errors::Unauthorized => e
+        Utility::Logger.error('Could not retrieve connectors metadata due to authorization error. Check debug logs for full error message.')
+        Utility::Logger.debug("Error: #{e.message}")
+
+        raise Utility::AuthorizationError.new, 'Unauthorized'
```

Comment:
I think it will work, but not always  this response can be actually returned by any method that calls Elasticsearch. I think we should handle it in more generic way, catching this error more widely.